test/src/materials/ADCoupledMaterial.C

```diff
@@ -0,0 +1,45 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+#include "ADCoupledMaterial.h"
+
+registerADMooseObject("MooseTestApp", ADCoupledMaterial);
+
+defineADValidParams(
+    ADCoupledMaterial, ADMaterial, params.addRequiredCoupledVar("coupled_var", "A coupledvariable");
+    params.addRequiredParam<MaterialPropertyName>("ad_mat_prop",
+                                                  "Name of the ad property this material defines");
+    params.addRequiredParam<MaterialPropertyName>(
+        "regular_mat_prop", "Name of the regular property this material defines"););
+
+template <ComputeStage compute_stage>
+ADCoupledMaterial<compute_stage>::ADCoupledMaterial(const InputParameters & parameters)
+  : ADMaterial<compute_stage>(parameters),
+    _ad_mat_prop(this->template declareADProperty<Real>(
```

Comment:
Yikes.  I'd like to hide the `this->template declareADProperty` behind a macro called `declareADProperty` (yest, that will mean that the actual function name will need to be something slightly different